### PR TITLE
Konsti trace regularization

### PR DESCRIPTION
**Pull request by KonstiNik**
Add regularization to the training of neural networks. 
There are two regularizers available:
- Norm regularizer, which penalizes the weights using a (user-defined) norm function. 
- Trace regularizer, which penalizes the gradients of the weights. 
Each regularizer comes with the option of using a schedule, that scales the effect of the regularization.

### CODE REVIEW

File: CI/unit_tests/regulaizers/test_norm_regularizer.py [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+--------
+If you use this module please cite us with:
+
+Summary
+-------
+Test the norm regularizer class.
+"""
+import os
+
+os.environ["CUDA_VISIBLE_DEVICES"] = "-1"
```
Comment (SamTov) [inline CI/unit_tests/regulaizers/test_norm_regularizer.py:30]: If we can avoid it, it would be better not to have this here.
